### PR TITLE
fixes module import(s)

### PR DESCRIPTION
**Pull request by dkress59**
& adds E2E-ish test workflow

### CODE REVIEW

**Comment by dkress59** [comment]
so basically adding a index.d.ts in the root dir and including dist in the published package did the trick :relieved:

**Comment by dkress59** [comment]
hm, for some reason sonar.exclusions stopped working after commit 76a1785 :-1:

**Comment by dkress59** [comment]
> hm, for some reason sonar.exclusions stopped working after commit [76a1785](https://github.com/react-async-storage/react-async-storage/commit/76a17859ddb9053c0351e488658871e7d40f189a) -1

I monkey patched this by adding `dist/` to .gitignore – `sonar.excludes` and `sonar.coverage.excludes` aren't working from the sonar-project.properties – I suggest we add the `dist` folder to the excludes via the SonarCloud UI (reminder: the compiled `dist` dir needs to be published for a valid module/entry point).

Other than that the module import is now fixed :smile:

**Comment by dkress59** [comment]
I should have taken a closer look at the npm scripts a bit earlier (awkward): I noticed the prepublishOnly script, so now I cleaned things up and this PR actually is only about the (now also improved) CRA install test!

File: install-test.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { StorageProvider } from 'react-async-storage'
+import React, { useEffect, useState } from 'react'
+
+const StorageComponent = () => {
+    const [test, setTest] = useState<null | string>(null)
+    useEffect(() => {
+        const ls = localStorage.getItem('test')
+        setTest(ls)
```
Comment (Goldziher) [inline install-test.tsx:8]: So this is basically wrong 😃. 

To test the storage engine, you need to use the storage engine - not localStorage.

Import the useStorage hook inside the component, then use an async call to getItem. You can see this used in the tests.

File: package.json [hunk 2]
```diff
@@ -28,14 +28,15 @@
     "unpkg": "dist/umd/index.min.js",
     "types": "dist/types/index.d.ts",
     "scripts": {
-        "build": "rimraf dist && rollup -c rollup.config.js && prettier --write './dist/**/*.ts'",
```
Comment (Goldziher) [inline package.json:31]: The point was to make the typings nice looking

File: install-test.tsx (hunk 1 above)
Comment (dkress59) [reply install-test.tsx:8]: haha, good to know! I'll adjust accordingly, but I think you get that it's more about the import statement at the top

File: package.json (hunk 2 above)
Comment (dkress59) [reply package.json:31]: ok now I see

**Pull request closed without merging**